### PR TITLE
YLP-488 HCP Profile dialog

### PR DESCRIPTION
**Pull request by leovalette**
There is no need to pay attention to js files. There was both double quotes and single quotes so it has been homogenized by VSCode when I modified them

### CODE REVIEW

File: packages/yourloops/pages/hcp/profile-dialog.tsx [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+          value={hbA1c}
+          className={textField}
+          InputProps={{ classes: { disabled } }}
+        />
+        <TextField
+          fullWidth
+          disabled
+          id="units"
+          label={t("units")}
+          value={user.settings?.units?.bg || Units.gram}
```
Comment (yvon-dblg) [inline packages/yourloops/pages/hcp/profile-dialog.tsx:136]:
```suggestion
          value={user.settings?.units?.bg ?? Units.gram}
```
We can use this specific operator now

File: packages/yourloops/pages/hcp/profile-dialog.tsx [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+          label={t("lastName")}
+          value={user.profile?.lastName}
+          className={textField}
+          InputProps={{ classes: { disabled } }}
+        />
+        <TextField
+          fullWidth
+          disabled
+          id="birthDate"
+          label={t("hcp-patient-profile-birthdate")}
```
Comment (yvon-dblg) [inline packages/yourloops/pages/hcp/profile-dialog.tsx:108]:
```suggestion
          label={t("birthdate")}
```
?

File: locales/en/yourloops.json [hunk 3]
```diff
@@ -104,6 +104,7 @@
   "team-edit-dialog-placeholder-name": "Name",
   "team-edit-dialog-placeholder-phone": "Phone",
   "Cancel": "Cancel",
+  "common-close": "Close",
```
Comment (yvon-dblg) [inline locales/en/yourloops.json:107]:
```suggestion
  "close": "Close",
```
?

File: packages/yourloops/pages/hcp/profile-dialog.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+          id="units"
+          label={t("units")}
+          value={user.settings?.units?.bg || Units.gram}
+          className={textField}
+          InputProps={{ classes: { disabled } }}
+        />
+      </DialogContent>
+      <DialogActions>
+        <Button onClick={handleClose} color="primary">
+          {t("common-close")}
```
Comment (yvon-dblg) [inline packages/yourloops/pages/hcp/profile-dialog.tsx:143]:
```suggestion
          {t("close")}
```
?

File: packages/yourloops/pages/hcp/profile-dialog.tsx [hunk 5]
```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2021, Diabeloop
+ * HCPs patient modale
```
Comment (yvon-dblg) [inline packages/yourloops/pages/hcp/profile-dialog.tsx:3]:
```suggestion
 * patient profile modale
```

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
- I would have set the lastName to fullName if missing first/last name
image

- Refactor the blip code for this PR seems unnecessary, too much line are modified, it's out of scope for me.
  - I would have a specific commit for, or better another PR
- Birthday: The date should be localized, if not available, an explicit message should be displayed.
- I think `packages/yourloops/pages/hcp/profile-dialog.tsx` should be in the `components` directory. Since it's a component used outside the /hcp pages.

*Out of scope (but since I saw the change here):*
- notif-hcp-invitation / notif-caregiver-invitation : placeholders are {{name}} and not {name}

File: packages/yourloops/pages/hcp/profile-dialog.tsx (hunk 2 above)
Comment (leovalette) [reply packages/yourloops/pages/hcp/profile-dialog.tsx:108]: birthdate key doesn't exist

File: locales/en/yourloops.json (hunk 3 above)
Comment (leovalette) [reply locales/en/yourloops.json:107]: As you wish

File: packages/yourloops/components/profile-dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import moment from "moment";
```
[the remark this answers is not in the document]
Comment (yvon-dblg) [reply packages/yourloops/components/profile-dialog.tsx:29]: We only have declared `moment-timezone` in the `package.json`.

Also, moment is now a "deprecated" (low maintenance) project.
We should probably use another one (see https://momentjs.com/docs/#/-project-status/recommendations/)

**Review by yvon-dblg — CHANGES_REQUESTED** [review]
- The pb with fullName only in the profile is still present.
- A few remarks below.

File: packages/yourloops/components/profile-dialog.tsx [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+  })
+);
+
+const ProfileDialog: FunctionComponent<ProfileDialogProps> = ({ user, isOpen, handleClose }: ProfileDialogProps) => {
+  const { t } = useTranslation("yourloops");
+  const { textField, title, disabled } = useStyles();
+
+  const mail = user?.emails ? user.emails[0] : "";
+  const hbA1c = "8.5%"; // TODO
+  const birthDate = moment.utc(user?.profile?.patient?.birthday).format("L");
```
[the remark this answers is not in the document]
Comment (leovalette) [reply packages/yourloops/components/profile-dialog.tsx:80]: Let's say we'll change the lib in a dedicated PR

File: packages/yourloops/components/profile-dialog.tsx (hunk 6 above)
[the remark this answers is not in the document]
Comment (leovalette) [reply packages/yourloops/components/profile-dialog.tsx:29]: Let's say we'll change the lib in a dedicated PR

File: packages/yourloops/components/profile-dialog.tsx (hunk 7 above)
[the remark this answers is not in the document]
Comment (leovalette) [reply packages/yourloops/components/profile-dialog.tsx:80]: "L" is already translating

File: locales/en/yourloops.json (hunk 3 above)
Comment (yvon-dblg) [reply locales/en/yourloops.json]: On your other PR for de-duplicate the translation keys, I finally change my mind.
I will have probably set to "button-close" I think.
Well, maybe it's better to wait we have all the base code done to clean-up theses.

**Review by yvon-dblg — APPROVED** [review]
LGTM

Question: Do we merge theses PR
- https://github.com/mdblp/blip/pull/398
- https://github.com/mdblp/blip/pull/399
before or after yours ?

**Comment by leovalette** [comment]
> LGTM
> 
> Question: Do we merge theses PR
> 
> * #398
> * #399
>   before or after yours ?

Doesn't matter to me, I'll rebase if needed

**Pull request merged**